spec/tenPinSpec.js

```diff
@@ -0,0 +1,172 @@
+describe('TenPin', function() {
+
+  beforeEach(function(){
+    frame = new Frame();
+    tenP = new TenPin(Frame);
+    roll1 = new Roll;
+    roll2 = new Roll;
+    rollBonus = new Roll;
+  });
+
+  describe('game begins.', function() {
+
+    it('The final frame variable is set to fasle', function() {
+      expect(tenP.finalFrame).toEqual(false)
+    });
+
+    it('Sets the current frame to 0', function() {
+      expect(tenP.currentFrame).toEqual(0)
+    });
+
+     it('Sets the max frame to 10', function() {
+      expect(tenP.maxFrame).toEqual(10)
+    });
+
+    it('Creates the first frame object in the framesCreated array', function() {
+      expect(tenP.framesCreated.length).toEqual(1)
+    });
+
+    it('Once a frame is played it sets the current frame to the next frame', function() {
+      tenP.createMoveFrame()
+      expect(tenP.currentFrame).toEqual(1)
+    });
+
+     it('Once a frame is played it creates the next frame object', function() {
+      tenP.createMoveFrame()
+      expect(tenP.framesCreated.length).toEqual(2)
+    });
+
+  });
+
+  describe('final frame.', function() {
+
+    it('Knows that frame 9 is not the final frame', function() {
+      tenP.currentFrame = 8
+      expect(tenP.isFinalFrame()).toEqual(false)
+    });
+
+    it('Knows that when in the max frame it is the final frame', function() {
+      tenP.currentFrame = 9
+      expect(tenP.isFinalFrame()).toEqual(true)
+    });
+
+  })
+
+  describe('frame is not a strike or spare.', function() {
+
+    it('Scores the first roll of 7 for the frame 1', function() {
+      //trouble setting a spyOn to see that an object in the framesCreated array was alled. Any pointers would be great
+      spyOn(frame, 'setRollOneScore')
+      tenP.frameFirstRoll(roll1)
+      expect(frame.setRollOneScore()).toHaveBeenCalled()
+    });
+
+    it('Scores the second roll of 2 for the frame 1', function() {
+      spyOn(frame, 'setRollTwoScore')
+      tenP.frameSecondRoll(roll2)
+      expect(frame.setRollTwoScore()).toHaveBeenCalled()
+    });
```

Comment:
spies utilise method call without arguments, so should be frame.setRollTwoScore instead of frame.setRollTwoScore()